### PR TITLE
Fix security issue with XML parsing

### PR DESCRIPTION
**Pull request by mvafin**
Root cause analysis: `xml.etree.ElementTree.parse` is unsafe to use, `xml.etree.ElementTree` is unsafe to include.

Solution: Replace unsafe `xml.etree.ElementTree.parse` with safe `defusedxml.ElementTree.parse`. Use `defusedxml.defuse_sdtlib()` to patch `xml.etree.ElementTree` and use objects from patched library.

Ticket: 55711

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: N/A

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Comment by rkazants** [comment]
@mvafin, @asomsiko, why should we care of this if the MO tool is offline tool? We can break something in MO by changing this logic that is more risky.

File: model-optimizer/mo/utils/ir_engine/ir_engine.py [hunk 1]
```diff
@@ -50,7 +57,8 @@ def __load_xml(self):
         self.graph = Graph()
         self.graph.graph['hashes'] = {}
 
-        self.graph.graph['ir_version'] = int(xml_root.attrib['version']) if xml_root.attrib.get('version') is not None else None
+        self.graph.graph['ir_version'] = int(xml_root.attrib['version']) if xml_root.attrib.get(
+            'version') is not None else None
```
Comment (rkazants) [inline model-optimizer/mo/utils/ir_engine/ir_engine.py:61]: please return the previous version. Both seem to be the same and the first one is better

File: model-optimizer/mo/utils/ir_engine/ir_engine.py [hunk 2]
```diff
@@ -206,8 +213,11 @@ def __load_layer(self, layer):
                 new_attrs = self.__normalize_attrs(attr.attrib)
                 if layer.attrib['type'] == 'Const':
                     assert 'offset' in new_attrs and 'size' in new_attrs, \
-                        'Incorrect attributes for Const layer, {} instead of {}!'.format(new_attrs.keys(), ['offset', 'size'])
-                    new_attrs.update(self.__prepare_bin_attrs(layer, 0, 'custom', new_attrs['offset'], new_attrs['size'], layer[1][0].attrib['precision']))
+                        'Incorrect attributes for Const layer, {} instead of {}!'.format(new_attrs.keys(),
+                                                                                         ['offset', 'size'])
+                    new_attrs.update(
+                        self.__prepare_bin_attrs(layer, 0, 'custom', new_attrs['offset'], new_attrs['size'],
+                                                 layer[1][0].attrib['precision']))
```
Comment (rkazants) [inline model-optimizer/mo/utils/ir_engine/ir_engine.py:220]: please return the previous version. the same comment as above.

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py [hunk 3]
```diff
@@ -12,6 +16,8 @@
 from mo.utils.utils import refer_to_faq_msg
 from mo.utils.version import get_version
 
+defuse_stdlib()
```
Comment (rkazants) [inline model-optimizer/mo/back/ie_ir_ver_2/emitter.py:19]: could you please explain what this routine is doing and why isn't it placed inside function?
Comment may be helpful

Comment (mvafin) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py]: Reworked for a better solution

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py [hunk 4]
```diff
@@ -12,6 +13,8 @@
 from mo.utils.utils import refer_to_faq_msg
 from mo.utils.version import get_version
 
+ET_defused = defuse_stdlib()[ET]
```
Comment (rkazants) [inline model-optimizer/mo/back/ie_ir_ver_2/emitter.py:16]: still no comment. Please add it

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py [hunk 5]
```diff
@@ -2,8 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import hashlib
-from xml.etree.ElementTree import Element, SubElement, tostring, ElementTree
 
+from defusedxml import defuse_stdlib
+import defusedxml.ElementTree as ET
```
Comment (rkazants) [inline model-optimizer/mo/back/ie_ir_ver_2/emitter.py:7]: why can't we do:
'from defusedxml.ElementTreeimport Element, ...'
so there will be no need to change code below.

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py (hunk 4 above)
Comment (rkazants) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:16]: can we insert it into import block and do needed imports like:
'from ET_defused import Element, ...' so there will be no need to change the code below

File: model-optimizer/mo/middle/passes/tensor_names.py [hunk 6]
```diff
@@ -1,9 +1,9 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-from xml.etree.ElementTree import Element, SubElement, tostring
-
+from defusedxml import defuse_stdlib
 from defusedxml.minidom import parseString
+import defusedxml.ElementTree as ET
```
Comment (rkazants) [inline model-optimizer/mo/middle/passes/tensor_names.py:6]: the same comment as for the previous file

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py (hunk 5 above)
Comment (mvafin) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:7]: There is no Element, SubElement, ElementTree or tostring in defusedxml.ElementTree

Comment (rkazants) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:7]: from ET_defused import Element, ...

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py (hunk 4 above)
Comment (mvafin) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:18]: We cant import from defuse_stdlib, but we can assign everything in the beginning of file.

**Review by rkazants — APPROVED** [review]

**Review by achetver — APPROVED** [review]

**Review by evolosen — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**